guides/unittesting-django-wagtail.md

```diff
@@ -0,0 +1,451 @@
+
+```python
+    def test_flags_index(self):
+        response = self.client.get('/admin/flags/')
+        self.assertEqual(response.status_code, 200)
+```
+
+We generally recommend using `WagtailTestUtils` to login and test the admin unless specific user properties or permissions are needed for specific tests.
+
+### Django management commands
+
+When testing custom management commands [Django provides a `call_command()` function](https://docs.djangoproject.com/en/2.1/topics/testing/tools/#management-commands) which will call the command direct the output into a `StringIO` object to be introspected. 
+
+For example, our [custom makemessages command (that adds support for Jinja2 templates)](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/management/commands/makemessages.py) is tested [using `call_command()`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/tests/management/commands/test_makemessages.py#L66) (although it inspects the resulting `.po` file):
+
+```python
+from django.core.management import call_command
+from django.test import SimpleTestCase
+
+class TestCustomMakeMessages(SimpleTestCase):
+    def test_extraction_works_as_expected_including_jinja2_block(self):
+        call_command('makemessages', locale=[self.LOCALE], verbosity=0)
+
+        with open(self.PO_FILE, 'r') as f:
+            contents = f.read()
+
+        expected = '''...'''
+        self.assertIn(expected, contents)
+```
+
+### Feature-flagged code
+
+We use [Django-Flags](https://cfpb.github.io/django-flags/) to feature flag code that should be run under certain conditions, such as waiting for a particular launch window or to A/B test. Sometimes it is necessary to test that code is actually flagged. 
+
+In all cases, the easiest way to test with explicit flag states is to [override the `FLAGS` setting](#overriding-settings) and include only your specific flag with a `boolean` condition of `True`.
+
+For example, to test [a function that serves a URL via Wagtail depending on the state of a flag](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/urls.py#L42-L53), the base URL tests [check behavior with a flag explicitly enabled and again with that flag explicitly disabled](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/tests/test_urls.py#L92-L109): 
+
+```python
+class FlaggedWagtailOnlyViewTests(TestCase):
+
+    @override_settings(FLAGS={'MY_TEST_FLAG': {'boolean': True}})
+    def test_flag_set_returns_view_that_calls_wagtail_serve_view(self):
+        response = self.client.get('/')
+        self.assertContains(
+            response,
+            'U.S. government agency that makes sure banks'
+        )
+
+    @override_settings(FLAGS={'MY_TEST_FLAG': {'boolean': False}})
+    def test_flag_not_set_returns_view_that_raises_http404(self):
+        response = self.client.get('/')
+        self.assertEqual(response.status_code, 404)
+```
+
+## Additional reading
```

Comment:
I might move this section up to the top of the file to provide a baseline level of understanding before going into specifics about how we do things.